python/aws_lambda_powertools/tracing/base.py

```diff
@@ -0,0 +1,233 @@
+from __future__ import annotations
+
+import abc
+import logging
+from typing import Any, List
+
+import aws_xray_sdk
+import aws_xray_sdk.core
+
+is_cold_start = True
+logger = logging.getLogger(__name__)
+
+
+class TracerProvider(metaclass=abc.ABCMeta):
+    """Tracer provider abstract class
+
+    Providers should be initialized independently. This
+    allows providers to control their config/initialization,
+    and only pass a class instance to
+    `aws_lambda_powertools.tracing.tracer.Tracer`.
+
+    It also allows custom providers to keep lean while Tracer provide:
+
+    * a simplified UX
+    * decorators for Lambda handler and methods
+    * auto-patching, patch all modules by default or a subset
+    * disabling all tracing operations with a single parameter or env var
+
+    Trace providers should implement the following methods:
+
+    * **patch**
```

Comment:
This implementations is inspired by how X-Ray does things. Have you though about how this will look for the most used tracer providers and if people won't have to shoehorn other concepts?

Reply:
totally true and hence why I went back and forth with this - It's great to hear a second opinion as I think it's best to drop the provider support altogether.